lib/board.rb

```diff
@@ -21,20 +22,26 @@ def valid_coordinate?(coordinate)
   end
 
   def valid_placement?(ship, coordinates)
-    if coordinates.length != ship.length
+    if overlap?(coordinates) == true
```

Comment:
put the overlap method at the beginning. I think it could work anywhere within the method

Reply:
Yeah, placement shouldn't be an issue with methods in classes, because they are only read when called upon.